[RobotWizzard] iP

RobotWizzard opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

RobotWizzard committed:
rename file "Duke.java" to "Bob.java"

RobotWizzard committed:
change logo

RobotWizzard committed:
add say function and greeting

RobotWizzard committed:
add exit

RobotWizzard committed:
change say() function to have space before each newline

RobotWizzard committed:
add inputs, use enums for commands

RobotWizzard committed:
add echo command, add indentations

RobotWizzard committed:
add text storage, add list command

RobotWizzard committed:
add Task class

RobotWizzard committed:
improve command handling, add "add" command

RobotWizzard committed:
add mark and unmark commands

RobotWizzard committed:
add task types (Todo, Deadline, Event)

RobotWizzard committed:
add commands for adding task types

RobotWizzard committed:
fix list command for empty list

RobotWizzard committed:
Add inputs and expected output for automated text ui testing

RobotWizzard committed:
add 'BobException', 'MissingArgumentException', 'UnknownCommandException'

RobotWizzard committed:
add 'IncorrectArgumentException', add handling for incorrect arguments when running the 'mark' and 'unmark' commands

RobotWizzard committed:
add automated ui testing for exceptions handling; fix extra whitespace when parsing commands with arguments; fix incorrect throwing of exception for the 'mark' and 'unmark' commands

RobotWizzard committed:
add 'delete' command

LowXiSi commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,207 @@
+        System.out.println(logo);
+        greet();
+        while (true) {
+            boolean executed = false;
```
Should executed be named as isExecuted instead to show that it is a boolean value?

LowXiSi commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,207 @@
+    private static final String LINE_PREFIX = "    ";
+
+    private static String argument = "";
+    private static final List<Task> list = new ArrayList<>();
```
Should list be named as tasks instead to represent a collection of tasks?

LowXiSi reviewed (commented):
Overall, most of your code seems to follow the coding standard! :)

potaotototo commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,207 @@
+                    StringBuilder text = new StringBuilder();
+                    int i;
+                    for (i = 0; i < list.size() - 1; ++i) {
+                        text.append(i + 1).append(".").append(list.get(i)).append("\n");
```
Is there anyway that this could be abstracted? For example, maybe a method inside the Task class or its inheriting classes?

potaotototo commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,207 @@
+            }
+
+            private static Task getTask(int fromIndex, int toIndex) {
+                String desc = fromIndex < toIndex
+                                ? argument.substring(0, fromIndex)
+                                : argument.substring(0, toIndex);
```
Not sure whether this is accurate? Maybe try to add comments to explain why you choose to incorporate fromIndex and toIndex?

potaotototo commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,207 @@
+                }
+                int i;
+                try {
+                    i = Integer.parseInt(argument) - 1;
```
Not compulsory but could try to name i as something else. This can help other programmers understand the code easily.

potaotototo commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,207 @@
+            }
+        };
+
+        public final String CMD;
```
I think the coding standards mentioned that we should avoid short forms? Maybe can name as COMMAND instead of CMD?

RobotWizzard committed:
add "Storage" class to handle data saving and loading

RobotWizzard committed:
add "FileCorruptedException" for basic handling of file corruption

RobotWizzard committed:
Merge branch 'branch-Level-7'

RobotWizzard committed:
add basic handling of date time (parsing, formatting, encoding, decoding)

RobotWizzard committed:
change time and year to be optional

RobotWizzard committed:
add exception handling for invalid datetime input

RobotWizzard committed:
add special values "now" and "tmr" when parsing datetime

RobotWizzard committed:
Merge branch 'branch-Level-8'

RobotWizzard committed:
put all classes into package "bob"

RobotWizzard committed:
put exceptions into sub-package "exception"; fix shell script for testing

RobotWizzard committed:
put tasks in sub-package "task"

RobotWizzard committed:
fix issues with protected accessor in tasks

RobotWizzard committed:
Merge branch 'branch-A-Packages'

RobotWizzard committed:
Merge branch 'master' into branch-A-MoreOOP

RobotWizzard committed:
extract Ui class to deal with user interactions

RobotWizzard committed:
extract TaskList class

RobotWizzard committed:
fix Storage not throwing exception when file is corrupted; add reset command

RobotWizzard committed:
extract commands into sub-package "command"

RobotWizzard committed:
define method of searching commands

RobotWizzard committed:
extract Parser class to deal with parsing input

RobotWizzard committed:
generalize Command argument handling via tokenization

RobotWizzard committed:
extract DataTime class to deal with parsing and formatting datetime

RobotWizzard committed:
make Bob methods non-static

RobotWizzard committed:
Merge branch 'add-gradle-support' of https://github.com/RobotWizzard/ip

RobotWizzard committed:
setup gradle for build automation

RobotWizzard committed:
add JUnit tests for DateTime and Parser

RobotWizzard committed:
extract ClassGetter class to deal with getting class

RobotWizzard committed:
add handling for getting classes when run in .jar

RobotWizzard committed:
add header comments for Bob, ClassGetter, DateTime

RobotWizzard committed:
add header comments for Parser

RobotWizzard committed:
add header comments for Storage

RobotWizzard committed:
add header comments for Ui

RobotWizzard committed:
add header comments for TaskList

RobotWizzard committed:
add header comments for Command

RobotWizzard committed:
add header comments for Task

RobotWizzard committed:
add header comments for all exceptions

RobotWizzard committed:
refine header comments for a better looking documentation website

RobotWizzard committed:
fix minor issues with header comments

RobotWizzard committed:
rename method in ClassGetter

RobotWizzard committed:
tweak the code to comply with the coding standard

RobotWizzard committed:
remove unused imports in DateTimeTest

RobotWizzard committed:
add FindCommand

RobotWizzard committed:
add find command to ParserTest

RobotWizzard committed:
add keyword highlighting for the find command

RobotWizzard committed:
fix wrong index being displayed for the find command

RobotWizzard committed:
Merge branch 'branch-A-JavaDoc'

RobotWizzard committed:
Merge branch-A-CodingStandard

RobotWizzard committed:
Merge branch-Level-9

RobotWizzard committed:
rename variable

RobotWizzard committed:
include javafx as a dependency

RobotWizzard committed:
add a simple non-functioning gui

RobotWizzard committed:
add simple styling to the gui

RobotWizzard committed:
add simple user input handling for gui

RobotWizzard committed:
modify user input handling to get response from Bob

RobotWizzard committed:
create MainWindow controller and view

RobotWizzard committed:
extract DialogBox view

RobotWizzard committed:
extract MainWindow controller from the Main class

RobotWizzard committed:
add auto-resizing elements

RobotWizzard committed:
add styling elements with css

RobotWizzard committed:
unbound scrollPane vValue with dialogContainer's height

RobotWizzard committed:
direct commands to the Bob class

RobotWizzard committed:
change style dynamically based on command

RobotWizzard committed:
add background image

RobotWizzard committed:
add special styling for error messages

RobotWizzard committed:
show greeting

RobotWizzard committed:
Merge branch 'branch-Level-10'

RobotWizzard committed:
add assertions throughout the code

Assertions provide additional safety by checking assumptions made within the code.

RobotWizzard committed:
improve code quality throughout the code

* Some methods have multiple levels of abstraction.
* Some header comments provide implementation details.

* Simplify Header comment for ClassGetter.getAllClassesJar by removing implementation details.
* Rename variable "i" in DeadlineCommand to "index" for more clarity.
* Extract FindCommand.getTasksAtIndices to convert list of found tasks to string.

Streams are used in ClassGetter.getAllClassesJar and FindCommand.getTasksAtIndices

RobotWizzard committed:
Merge pull request #2 from RobotWizzard/branch-A-CodeQuality

Improve code quality throughout the code

RobotWizzard committed:
Merge branch 'master' into branch-A-Assertions

RobotWizzard committed:
Merge pull request #3 from RobotWizzard/branch-A-Assertions

Add assertions throughout the code

RobotWizzard committed:
package DateTime and ClassGetter into bob.util

The project repository is getting messy.

Classifying classes into packages improves the code quality.

Let's package all the utility classes into bob.util.
Utility classes are defined as classes with only static methods.

RobotWizzard committed:
extract encoding and decoding to methods in Task

RobotWizzard committed:
fix UnmarkCommand not unmarking tasks bug

RobotWizzard committed:
Merge branch 'master' into extract-encode-decode

RobotWizzard committed:
make Storage.decode method more OOP

RobotWizzard committed:
package ClassGetter and DateTime into bob.util

The repository is getting a bit messy.

Let's extract the utility classes into their own package.
Utility classes are classes that only have static methods.

RobotWizzard committed:
add tag command and tags to tasks

RobotWizzard committed:
include tags in encoding

RobotWizzard committed:
merge with branch extract-encode-decode

RobotWizzard committed:
add print all tags to the "tag" command

RobotWizzard committed:
add print specific tag to the "tag" command

RobotWizzard committed:
add untag command

There are ways for users to tag tasks, but no way for them to remove
tags.

The "untag" command is added for users to untag specific tags
from a task, or all tags from a task.

RobotWizzard committed:
Merge branch 'branch-C-Tagging'

RobotWizzard committed:
extract string formatting as a FormattedString utility class

There is a need for classes to get the unformatted version of a string.
A FormattedString class would also enable greater flexibility when
formatting strings in the future, for example applying many formats to
a string.

Let's extract a FormattedString class that can apply red and
yellow-background to strings for now.
The FormattedString class acts like the StringBuilder class in Java
which allows chaining operations on it.

RobotWizzard committed:
add append method to FormattedString

There is currently no way to combine separate FormattedString objects.

Let's add an append method to FormattedString to allow appending other
FormattedString objects to it.

RobotWizzard committed:
add title and icon for the app

The app currently has the default icon and title, which shows at the top
of the window.

Adding a title and icon would improve the user experience.

Let's add a title and icon to our app.

RobotWizzard committed:
change the background image

The current background image does not work well as it stretches and
cannot be repeated, which does not look good.

Having a good background would improve the user experience.

Let's change the background to a repeating one.

RobotWizzard committed:
make the profile pictures circular

The app currently have square profile picture, which is different from
what users may want.

Let's make the profile pictures circular to
match the look of a chat app.

RobotWizzard committed:
change the colors of dialog boxes

The current dialog boxes have a background color that
stands out too much, and a border with multiple colors,
 which is too distracting.

A color theme that doesn't distract from the main actions
would improve user experience.

Let's change the color of the dialog boxes to be less distracting.

RobotWizzard committed:
add a simple user guide

New users will not know what are the features and commands of Bob.

A user guide will help new users orient themselves with Bob.

Let's add a new user guide that explains the features and
commands of Bob.

RobotWizzard committed:
change numbered lists to a code block in the user guide.

Numbered lists do not show up as expected in github pages.

Let's use code blocks in place of numbered lists instead.

RobotWizzard committed:
change code block in the details tag to br tags

Code blocks also do not show up as expected in github pages.

Let's try using <br> tags instead.

RobotWizzard committed:
change br tags in the details tag to use pre tag

While <br> tags do work in the details tag,
the list is not in a monospace font.

Let's try using the <pre> tag instead.

RobotWizzard committed:
Fix wrong command example for "untag" command

RobotWizzard committed:
Remove unused imports in UntagCommand

RobotWizzard committed:
Add header comments for DeadlineCommand

RobotWizzard committed:
Add header comments for DeleteCommand

RobotWizzard committed:
Add header comments for EventCommand

RobotWizzard committed:
Change expected command format in header comments to use <pre> tag
